src/Command/DoPopulateIndex.php

```diff
@@ -0,0 +1,169 @@
+
+    public function __construct(
+        private readonly IndexRepository $indexRepository,
+        private readonly DocumentRepository $documentRepository,
+        private readonly DocumentHelper $documentHelper,
+        private readonly ConfigurationRepository $configurationRepository,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setName(self::COMMAND_NAMESPACE . 'do-populate-index')
+            ->setHidden(true)
+            ->setDescription('[INTERNAL]')
+            ->addOption(self::OPTION_CONFIG, mode: InputOption::VALUE_REQUIRED)
+            ->addOption(self::OPTION_INDEX, mode: InputOption::VALUE_REQUIRED)
+            ->addOption(self::OPTION_BATCH_NUMBER, mode: InputOption::VALUE_REQUIRED)
+            ->addOption(self::OPTION_LISTING_COUNT, mode: InputOption::VALUE_REQUIRED)
+            ->addOption(self::OPTION_DOCUMENT, mode: InputOption::VALUE_REQUIRED)
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $indexConfig = $this->getIndex();
+
+        if (!$indexConfig instanceof IndexInterface) {
+            return self::FAILURE;
+        }
+
+        return $this->populateIndex($indexConfig, $indexConfig->getBlueGreenInactiveElasticaIndex());
+    }
+
+    private function getIndex(): ?IndexInterface
+    {
+        foreach ($this->indexRepository->flattenedAll() as $indexConfig) {
+            if ($indexConfig->getName() === $this->input->getOption(self::OPTION_CONFIG)) {
+                return $indexConfig;
+            }
+        }
+
+        return null;
+    }
+
+    private function populateIndex(IndexInterface $indexConfig, ElasticaIndex $esIndex): int
+    {
+        ProgressBar::setFormatDefinition('custom', "%percent%%\t%remaining%\t%memory%\n%message%");
+
+        $batchNumber = (int) $this->input->getOption(self::OPTION_BATCH_NUMBER);
+        $listingCount = (int) $this->input->getOption(self::OPTION_LISTING_COUNT);
+
+        $allowedDocuments = $indexConfig->getAllowedDocuments();
+        $document = $this->input->getOption(self::OPTION_DOCUMENT);
+
+        if (!in_array($document, $allowedDocuments, true)) {
+            return self::FAILURE;
+        }
+
+        $progressBar = new ProgressBar($this->output, $listingCount > 0 ? $listingCount : 1);
+        $progressBar->setMessage($document);
+        $progressBar->setFormat('custom');
+        $progressBar->setProgress($batchNumber * $indexConfig->getBatchSize());
+
+        if (!$indexConfig->shouldIndexInSubprocesses()) {
+            $numberOfBatches = ceil($listingCount / $indexConfig->getBatchSize());
+
+            for ($batch = 0; $batch < $numberOfBatches; $batch++) {
+                $exitCode = $this->doPopulateIndex($esIndex, $indexConfig, $progressBar, $document, $batch);
+
+                if ($exitCode !== self::SUCCESS) {
+                    return $exitCode;
+                }
```

Comment:
```suggestion
                return $this->doPopulateIndex($esIndex, $indexConfig, $progressBar, $document, $batch);
```

In essence you always return `$exitCode`, either explicitly (when != success) or implicitly on line 105.

Reply:
👍